### PR TITLE
Avoid division by zero

### LINKED ISSUE #13467: WSOD after importing pull request #13463
Some time after git pull from https://github.com/friendica/friendica/commit/dd7bea4bd146acf52086d303fce34a83d52f2516 I have a WSOD and the following error message in the log.

```
2023-09-24T14:34:31Z ERROR app Uncaught Exception DivisionByZeroError: "Division by zero" at /var/www/html/src/Model/Post/Media.php line 1181

DivisionByZeroError: Division by zero at /var/www/html/src/Model/Post/Media.php:1181
Stack trace:
#0 /var/www/html/src/Model/Item.php(3539): Friendica\Model\Post\Media::getAllocatedHeightByMedia()
#1 /var/www/html/src/Model/Item.php(3246): Friendica\Model\Item::addVisualAttachments().
#2 /var/www/html/src/Object/Post.php(431): Friendica\Model\Item::prepareBody()
#3 /var/www/html/src/Object/Thread.php(204): Friendica\Object\Post->getTemplateData().
#4 /var/www/html/src/Content/Conversation.php(685): Friendica\Object\Post->getTemplateData()
#5 /var/www/html/src/Content/Conversation.php(583): Friendica\Object\Conversation->getThreadList()
#6 /var/www/html/src/Module/Conversation/Network.php(239): Friendica\Content\Conversation->render()
#7 /var/www/html/src/BaseModule.php(244): Friendica\Module\Conversation\Network->content().
#8 /var/www/html/src/App.php(710): Friendica\BaseModule->run()
#9 /var/www/html/index.php(59): Friendica\App->runFrontend()
#10 {main}
```

by a 
`git checkout ce16cd12ec46413581f5ade6487387f56723e3`
I was able to fix the WSOD 

### Platform Info

Friendica Version:

Friendica Source: 2023.09-dev - 1534

PHP version: 8.1.2

SQL version: 10.6.12

### PR DESCRIPTION
**Pull request by annando**
Fixes https://github.com/friendica/friendica/issues/13467

### CODE REVIEW

File: src/Model/Post/Media.php [hunk 1]
```diff
@@ -1178,6 +1178,12 @@ public static function getUrlForId(int $id, string $size = ''): string
...
 	public static function getAllocatedHeightByMedia(array $media): string
 	{
-		return (100 * $media['height'] / $media['width']) . '%';
+		if (!empty($media['height']) && !empty($media['width'])) {
+			return (100 * $media['height'] / $media['width']) . '%';
+		}
+		if (!empty($media['preview-height']) && !empty($media['preview-width'])) {
+			return (100 * $media['preview-height'] / $media['preview-width']) . '%';
+		}
+		return "100%";
```
Comment (MrPetovan) [inline src/Model/Post/Media.php:1187]: This means the image is a square, which isn’t necessarily the case. How come some saved widths are `0` in the first place?

File: src/Model/Post/Media.php [hunk 2]
```diff
@@ -1178,6 +1178,12 @@ public static function getUrlForId(int $id, string $size = ''): string
 	 */
 	public static function getAllocatedHeightByMedia(array $media): string
 	{
-		return (100 * $media['height'] / $media['width']) . '%';
+		if (!empty($media['height']) && !empty($media['width'])) {
+			return (100 * $media['height'] / $media['width']) . '%';
+		}
+		if (!empty($media['preview-height']) && !empty($media['preview-width'])) {
```
Comment (MrPetovan) [inline src/Model/Post/Media.php:1184]: What’s the case where `width` and `height` aren’t filled but `preview-width` and `preview-height` are?

**Comment by MrPetovan** [comment]
Thank you for looking into this, however my feature assumed all `post-media` images had a `width` and `height` set. It was the case during my development and during the time it ran on my production node, but querying the `post-media` table directly after I saw this issue showed that my assumption was wrong.

The only corrective move we can do at the moment is completely pulling the feature that was based on a wrong assumption (https://github.com/friendica/friendica/pull/13469), and hopefully come back with another one that takes this uncertainty into account.

File: src/Model/Post/Media.php (hunk 2 above)
Comment (annando) [reply src/Model/Post/Media.php:1184]: I don't know. But I saw that where the function is called, we are using the preview and the none preview value as well: https://github.com/friendica/friendica/blob/dd7bea4bd146acf52086d303fce34a83d52f2516/src/Model/Item.php#L3364

File: src/Model/Post/Media.php (hunk 1 above)
Comment (annando) [reply src/Model/Post/Media.php:1187]: This ix just to prevent a division by zero.

**Comment by MrPetovan** [comment]
Invalidated by #13469

**Pull request closed without merging**